src/Enhavo/Bundle/AppBundle/Viewer/TableViewer.php

```diff
@@ -150,6 +164,100 @@ public function getMoveToPageRoute()
+        }
+        return $this->batch_actions;
+    }
+
+    protected function parseBatchActions()
+    {
+        $authorizationChecker = $this->container->get('security.authorization_checker');
+
+        $batch_actions = $this->getConfig()->get('table.batch_actions');
+        if (!$batch_actions) {
+            return array();
+        }
+
+        if (count($batch_actions) > 0) {
+            if (!isset($batch_actions[self::BATCH_ACTION_NONE_SELECTED])) {
+                $batch_actions[self::BATCH_ACTION_NONE_SELECTED] = array(
+                    'label'                 => 'table.batch.noneSelected.label',
+                    'confirm_message'       => '',
+                    'translation_domain'    => 'EnhavoAppBundle'
+                );
+            }
+
+            $translator = $this->container->get('translator');
+            $slugifier = new Slugifier();
+            $pos = 100;
+            $batch_actions_parsed = array();
+            foreach($batch_actions as $command => $value) {
+                if (isset($value['permission']) && $value['permission']) {
+                    if (!$authorizationChecker->isGranted($value['permission'])) {
+                        continue;
+                    }
+                }
+
+                $command_parsed = $slugifier->slugify($command);
+                $action_parsed = array();
+
+                if (isset($value['translation_domain']) && $value['translation_domain']) {
+                    $domain = $value['translation_domain'];
+                } else {
+                    $domain = $this->getTranslationDomain();
+                }
+
+                if (isset($value['label']) && $value['label']) {
+                    $action_parsed['label'] = $translator->trans($value['label'], array(), $domain);
+                } else {
+                    $action_parsed['label'] = $command;
+                }
+
+                if (isset($value['confirm_message']) && $value['confirm_message']) {
+                    $action_parsed['confirm_message'] = $translator->trans($value['confirm_message'], array(), $domain);
+                } else {
+                    $action_parsed['confirm_message'] = $translator->trans('table.batch.confirmMessageGeneric', array('%command%' => $command), 'EnhavoAppBundle');
+                }
+
+                if (isset($value['position']) && is_int($value['position']) && ($value['position'] >= 0)) {
+                    $action_parsed['position'] = $value['position'];
+                } else {
+                    if ($command == self::BATCH_ACTION_NONE_SELECTED) {
+                        $action_parsed['position'] = -1;
+                    } else {
+                        $action_parsed['position'] = $pos++;
+                    }
+                }
+                $batch_actions_parsed[$command_parsed] = $action_parsed;
+            }
+
+            uasort($batch_actions_parsed, array($this, 'cmp'));
```

Comment:
I think its better to use here an anonym function. Don't look like we will use the `cmp` again, and a class function seems like to heavy for that use case.